yarn-deployment/yarn_dbt.py

```diff
@@ -65,22 +66,50 @@ def main():
 # load the environment variables from yarn.env file
 def load_fetch_environment_variables():
     logging.info("Loading environment variables.")
-    dot_env_path = os.path.join(os.path.expanduser('~'), "yarn.env")
+    dot_env_path = os.path.join(os.path.expanduser("~"), "yarn.env")
     isFile = os.path.isfile(dot_env_path)
     if not isFile:
         logging.critical(
-            "Missing yarn.env file in home directory " + os.path.expanduser('~')
+            "Missing yarn.env file in home directory " + os.path.expanduser("~")
         )
         sys.exit(10)
 
     global ENV_VARIABLES
     ENV_VARIABLES = dotenv_values(dot_env_path)
     logging.info(f"Found config in %s", dot_env_path)
+    check_environment_variables(ENV_VARIABLES)
     for key, value in ENV_VARIABLES.items():
         logging.info(f"{key} : {value}")
     logging.info("Done Loading environment variables.")
 
 
+def check_environment_variables(ENV_VARIABLES):
+    expected_mandatory_keys = [
+        "DEPENDENCIES_PACKAGE_PATH_HDFS",
+        "DEPENDENCIES_PACKAGE_NAME",
+        "YARN_JAR",
+        "DBT_SERVICE_USER",
+        "DBT_PROJECT_NAME",
+        "YARN_RM_URI",
+        "DBT_HEADLESS_KEYTAB",
+        "DBT_HEADLESS_PRINCIPAL",
+        "CURRENT_DBT_USER",
+    ]
+
+    missing_keys = [key for key in expected_mandatory_keys if key not in ENV_VARIABLES]
+
+    # log and exit if yarn.env file doesn't have necessary key values
+    if missing_keys:
+        logging.critical("Missing keys in yarn.env: %s", missing_keys)
+        sys.exit(10)
+
+    # Below keys are optional. Set values incase user doesn't add these values.
+    ENV_VARIABLES.setdefault("DBT_DOCS_PORT", "7777")
+    ENV_VARIABLES.setdefault("YARN_CONTAINER_MEMORY", "2048")
+    ENV_VARIABLES.setdefault("YARN_TIMEOUT", "1800000")
+    ENV_VARIABLES.setdefault("APPLICATION_TAGS", "test-dbt")
```

Comment:
nitpicking : would "yarn-dbt" be better APPLICATION_TAGS

Reply:
Updated.